src/main/java/uk/gov/hmcts/reform/rolecategorymigration/Application.java

```diff
@@ -0,0 +1,127 @@
+
+	private final List<Pattern> roleCategoryPatterns = new ArrayList<>();
+	private final String ExceptionLabel = "*EXCEPTION*";
+
+	@Autowired
+	private ApplicationContext context;
+
+	@Autowired
+	private CaseDataRepository caseDataRepository;
+
+	@Autowired
+	private IdamRepository idamRepository;
+
+	public static void main(String[] args) {
+		SpringApplication.run(Application.class, args);
+	}
+
+	@Override
+	public void run(String... args) {
+		log.info("Data migration task has begun");
+		long startTime = System.currentTimeMillis();
+		roleCategoryPatterns.add(RoleCategory.CITIZEN.getPattern());
+		roleCategoryPatterns.add(RoleCategory.JUDICIAL.getPattern());
+		roleCategoryPatterns.add(RoleCategory.PROFESSIONAL.getPattern());
+
+		log.info("Starting to populate null role categories");
+
+		populateNewRoleCategory();
+
+		long endTime = System.currentTimeMillis();
+
+		log.info("Completed data population task in " + (endTime - startTime) + " milliseconds");
+
+		SpringApplication.exit(context, () -> 0);
+	}
+
+	private void populateNewRoleCategory() {
+
+		Page<String> dataWithoutRoleCategory = caseDataRepository.findCaseUsersById(PageRequest.of(0, 5000));
+		if (dataWithoutRoleCategory.isEmpty()) {
+			return;
+		}
+		List<String> listOfUserIds = new ArrayList<>(dataWithoutRoleCategory.getContent());
+		Map<String, List<String>> rolesToUserIdMap = new ConcurrentHashMap<>();
+		ForkJoinPool customThreadPool = new ForkJoinPool(10);
+		try {
+			customThreadPool.submit(() -> listOfUserIds.parallelStream().forEach(userId -> {
+				try {
+					rolesToUserIdMap.put(userId, retrieveIdamRoles(userId));
+				} catch (Exception exception) {
+					caseDataRepository.updateRoleCategory(ExceptionLabel, userId);
+					log.error(exception.getMessage());
+				}
+			})).get();
+		} catch (Exception exception) {
+			log.error(exception.getMessage());
+		} finally {
+			customThreadPool.shutdown();
+		}
+
+		RoleCategory matchingCategory;
+
+		for (String userId : rolesToUserIdMap.keySet()) {
+			try {
+				matchingCategory = matchRoleCategory(rolesToUserIdMap.get(userId), userId);
+				caseDataRepository.updateRoleCategory(matchingCategory.getName(), userId);
+			} catch (Exception exception) {
+				log.error(exception.getMessage());
+			}
+		}
+		populateNewRoleCategory();
+	}
+
+	private List<String> retrieveIdamRoles(String userId) {
+		List<String> idamRoles;
+		idamRoles = idamRepository.getUserRoles(userId).getRoles();
+		return idamRoles;
+	}
+
+	private RoleCategory matchRoleCategory(List<String> roles, String userId) {
+
+		RoleCategory firstCategory = null;
+		for (String role : roles) {
+			RoleCategory newCategory = null;
+			for (Pattern pattern : roleCategoryPatterns) {
+				if (pattern.matcher(role).matches()) {
+					newCategory = RoleCategory.getEnumFromPattern(pattern);
+					if (firstCategory == null) {
+						firstCategory = newCategory;
+						break;
+					} else if (firstCategory != newCategory) {
```

Comment:
```suggestion
			for (RoleCategory category : RoleCategory.values()) {
				if (category.getPattern().matcher(role).matches()) {
					if (matchedCategory == null) {
						matchedCategory = category;
						break;
					} else if (matchedCategory != category) {
```
I think we can simplify slightly and remove the need for `roleCategoryPatterns` and `getEnumFromPattern()` completely?